c++/src/kj/async.c++

```diff
@@ -278,6 +267,16 @@ public:
     node->onReady(this);
   }
 
+  Own<Task> pop() {
+    KJ_IF_MAYBE(n, next) { n->get()->prev = prev; }
+    Own<Task> self = kj::mv(KJ_ASSERT_NONNULL(*prev));
+    KJ_ASSERT(self.get() == this);
+    *prev = kj::mv(next);
+    next = nullptr;
```

Comment:
Oh yeah, I wanted to ask about this line. It's superfluous right, because we just moved `next` and `Maybe` nullifies after a move I thought. Maybe just a `KJ_DASSERT(next == nullptr)`?

Reply:
Historically, a moved-from `Maybe` was not guaranteed to be null. We recently decided to make it so, after seeing one or two bugs go by that would have been prevented by this. But I'm not sure how I feel about people relying on it.

I'd just leave the code as it is. The compiler should be able to optimize out the redundant nullification anyway (not that it would matter if it didn't).